### PR TITLE
FIX: redirect to previous page when subscribing to multiple tags

### LINKED ISSUE #5600: Redirect user to the same page after multi tag subscription
I think we should redirect the user to the same url after multi tag subscription, instead of /subscriptions so that the user will remain on the current research note or wiki page. Implemented the same in the pr.

### PR DESCRIPTION
**Pull request by Uzay-G**
Fixes #5600 

I included a test to verify that the redirection was working correctly

Here is a gif showing the functional redirection:
redirection-for-subscribe

### CODE REVIEW

**Comment by Uzay-G** [comment]
What do you think @SidharthBansal?

**Comment by SidharthBansal** [comment]
check your screenshot. there is dashboard always

**Comment by Uzay-G** [comment]
It worked. If you try on publiclab.org, you will be redirected to /subscriptions. In this case it redirected to the page the user was on

**Comment by SidharthBansal** [comment]
Please don't try at PL.org
Your previous prs merged are not deployed yet
Does this makes sense?

**Comment by Uzay-G** [comment]
Hey @SidharthBansal, I'm not at home right now and can't record my screen but when I subscribe to several tags on the stable version of PL I am redirected to /subscriptions:
Capture

On the other hand, as you can see in the gif, with my changes, I am redirected to the previous page I was on.

**Comment by Uzay-G** [comment]
Hey! I made another gif where you can see it working:
redirection-for-subscribe gif

There is a problem with the flash though...

**Comment by SidharthBansal** [comment]
@jywarren kindly review this. It looks good to me.
But will like to review once more after you review. 
@Uzay-G thanks for this. We can move to MK and SWB tests now I guess

File: app/controllers/subscription_controller.rb [hunk 1]
```diff
@@ -117,57 +117,58 @@ def digest
   end
 
   def multiple_add
-    if !params[:tagnames] || params[:tagnames] == ''
+    return_to = params[:return_to] || "/subscriptions?_=" + Time.now.to_i.to_s
+    if params[:tagnames].blank?
       flash[:notice] = "Please enter tags for subscription in the url."
-      redirect_to "/subscriptions" + "?_=" + Time.now.to_i.to_s
```
Comment (SidharthBansal) [inline app/controllers/subscription_controller.rb]: Can you please see why git is showing many lines of change eventhough it is two line change?

File: test/functional/subscription_controller_test.rb [hunk 2]
```diff
@@ -55,4 +55,12 @@ def setup
     get :add, params: { type: 'tag', name: 'blog' }, xhr: true
     assert users(:bob).following(:blog)
   end
+  
+  test "should redirect properly when subscribing to multiple tags" do
```
Comment (SidharthBansal) [inline test/functional/subscription_controller_test.rb]: Great

File: app/views/tag/_tags_popover.html.erb [hunk 3]
```diff
@@ -1,6 +1,7 @@
 <div class="pop_content subscribe-btn">
   <p> Follow related tags: </p>
   <%= form_with remote: true, url: '/subscribe/multiple/tag' do |f| %>
+    <input type="hidden" name="return_to" class="subscription-popover-redirect" value="<%= request.fullpath %>" />
```
Comment (SidharthBansal) [inline app/views/tag/_tags_popover.html.erb]: Cool

File: app/controllers/subscription_controller.rb (hunk 1 above)
Comment (SidharthBansal) [reply app/controllers/subscription_controller.rb]: sorry 4 lines of change

**Comment by SidharthBansal** [comment]
Oh the code is inside the else part so it showing indented code snippets. 
Great well done!

**Comment by Uzay-G** [comment]
Thank you so much! I couldn't really work today but will be available tomorrow! :+1:

**Pull request merged**